controller/contour/service.go

```diff
@@ -0,0 +1,271 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package contour
+
+import (
+	"context"
+	"fmt"
+
+	operatorv1alpha1 "github.com/projectcontour/contour-operator/api/v1alpha1"
+	utilequality "github.com/projectcontour/contour-operator/util/equality"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	// xdsPort is the network port number of the XDS service.
+	xdsPort = 8001
+	// httpPort is the network port number of HTTP.
+	httpPort = 80
+	// httpsPort is the network port number of HTTPS.
+	httpsPort = 443
+	// awsLbBackendProtoAnnotation is a Service annotation that places the AWS ELB into
+	// "TCP" mode so that it does not do HTTP negotiation for HTTPS connections at the
+	// ELB edge. The downside of this is the remote IP address of all connections will
+	// appear to be the internal address of the ELB. See docs/proxy-proto.md for information
+	// about enabling the PROXY protocol on the ELB to recover the original remote IP address.
+	awsLbBackendProtoAnnotation = "service.beta.kubernetes.io/aws-load-balancer-backend-protocol"
```

Comment:
Let's raise an issue about whether this should be configured, auto-detected or something else.

Reply:
xref: https://github.com/projectcontour/contour-operator/issues/49